src/frontend/hooks/useSelectedMapStyle.ts

```diff
@@ -0,0 +1,63 @@
+import { useMapAvailability } from "./useMapAvailability";
+import { useMapStyles, DEFAULT_MAP_ID } from "./useMapStyles";
+
+// This map style is used if the user is offline and there is no custom offline
+// map installed. It contains basic land, country, river and lake data and is
+// generated by https://github.com/digidem/mapeo-offline-map
+const FALLBACK_STYLE_URL = "asset://offline-style.json" as const;
+
+type SelectedMapStyle =
+  | {
+      status: "loading";
+    }
+  | {
+      styleUrl: string;
+      status: "success";
+      isOfflineFallback: false;
+    }
+  | {
+      styleUrl: typeof FALLBACK_STYLE_URL;
+      status: "success";
+      isOfflineFallback: true;
+    };
+
+/**
+ * Get the user selected map style.
+ *
+ * @returns The `styleUrl` and `status`, which is loading while map style
+ * availability is being checked. If there are no offline maps available and the
+ * user is offline, then `styleUrl` will be `undefined`.
+ */
+export function useSelectedMapStyle(): SelectedMapStyle {
+  const { styles, status, selectedStyleId } = useMapStyles();
+  const onlineMapAvailability = useMapAvailability("online");
+
+  if (status === "loading" || onlineMapAvailability === "unknown") {
+    return { status: "loading" };
+  }
+
+  // We have a check in useMapStyles to ensure that selectedStyleId is in the
+  // styles array, but this is to make TS happy
+  const selectedStyle =
+    styles.find(style => style.id === selectedStyleId) || styles[0];
+
+  if (
+    onlineMapAvailability === "unavailable" &&
+    selectedStyle.id === DEFAULT_MAP_ID
```

Comment:
In the future we need something better here, since it assumes that the only "online" map is the default map. I have some ideas about how we might handle the fallback map in the mapserver rather than the client, it would be good to discuss at some point.